api/v1beta1/operatorpolicy_types.go

```diff
@@ -97,30 +64,66 @@ type StatusConfig struct {
 type OperatorPolicySpec struct {
 	Severity          policyv1.Severity          `json:"severity,omitempty"`          // low, medium, high
 	RemediationAction policyv1.RemediationAction `json:"remediationAction,omitempty"` // inform, enforce
-	ComplianceType    policyv1.ComplianceType    `json:"complianceType"`              // Compliant, NonCompliant
-	// OperatorGroup requires at least 1 of target namespaces, or label selectors
-	// to be set to scope member operators' namespaced permissions. If both are provided,
-	// only the target namespace will be used and the label selector will be omitted.
+	ComplianceType    policyv1.ComplianceType    `json:"complianceType"`              // musthave
+
+	// Include the name, namespace, and any `spec` fields for the OperatorGroup.
+	// For more info, see `kubectl explain operatorgroup.spec` or
+	// https://olm.operatorframework.io/docs/concepts/crds/operatorgroup/
+	// +kubebuilder:pruning:PreserveUnknownFields
 	// +optional
-	OperatorGroup *OperatorGroup `json:"operatorGroup,omitempty"`
-	// Subscription defines an Application that can be installed
+	OperatorGroup *runtime.RawExtension `json:"operatorGroup,omitempty"`
+
+	// Include the namespace, and any `spec` fields for the Subscription.
+	// For more info, see `kubectl explain subscription.spec` or
+	// https://olm.operatorframework.io/docs/concepts/crds/subscription/
 	// +kubebuilder:validation:Required
-	Subscription SubscriptionSpec `json:"subscription"`
+	// +kubebuilder:pruning:PreserveUnknownFields
+	Subscription runtime.RawExtension `json:"subscription"`
```

Comment:
Would it be possible to have Subscription be a custom struct with the name and namespace fields and have the subscription fields as `runtime.RawExtension`? I think using the `inline` JSON tag may make this work.

Reply:
I can check.

Reply:
I think this was a neat idea but I couldn't get it to work. I tried like this:
```golang
type Subscription struct {
	Namespace string `json:"namespace"`
	Spec runtime.RawExtension `json:",inline"`
}
// ...
func buildSubscription() {
// ...
	spec := new(operatorv1alpha1.SubscriptionSpec)

	err := json.Unmarshal(policy.Spec.Subscription.Spec.Raw, spec)
	if err != nil {
		return nil, fmt.Errorf("error unmarshalling subscription: %w", err)
	}
// ...
}
```

The `json.Unmarshal` always returned an error, it looks like the RawExtension is always empty.